orbit_predictor/predictors/base.py

```diff
@@ -279,6 +281,76 @@ def get_next_pass(self, location, when_utc=None, max_elevation_gt=5,
         else:
             raise NotReachable('Propagation limit date exceeded')
 
+    def eclipses_since(self, when_utc, limit_date=None):
+        """
+        An iterator that yields all eclipses start and end times between
+        when_utc and limit_date.
+
+        The next eclipse with a end strictly after when_utc will be returned,
+        possibly the current eclipse.
+        The last eclipse returned starts before limit_date, but it can end
+        strictly after limit_date.
+        No circular orbits are not supported, and will raise NotImplementedError.
+        """
+        def _get_illumination(t):
+            my_start = start + dt.timedelta(seconds=t)
+            result = get_satellite_minus_penumbra_verticals(
+                self.get_only_position(my_start),
+                my_start
+            )
+            return result
+
+        orbital_period_s = 60 / self.mean_motion * 2 * np.pi
+        # A third orbit is used as the base window of the search.
```

Comment:
```suggestion
        # A third of the orbit period is used as the base window of the search.
```

Although, perhaps we could use `self.get_eclipse_duration(when_utc)` instead?

Reply:
I think not. The use of a third of the orbit in this part of the window search is because we want the most large window that avoids the inclusion of both possible local minimums (the positive and negative). It is not related with the duration of the eclipse, but with the fact that we have one eclipse per orbit and (perhaps) one positive local minimum per orbit. Perhaps using the orbit / 4 would be more robust, but the search would go slow. In all the cases I tested this window never failed to find the eclipses. Take into account that a satellite with a LTAN at 6 AM, have moments with an eclipse duration of one minute! and in this part of the search, we want to use the larger window we can get.